### ISSUE TITLE
500 error on viewFeedback

### ISSUE THREAD

**Issue by XinyuZeng** [autolab/Autolab#1875]
**Your environment**
Let us know details such as 
- Operating system: `Linux iZ6we4o97z593weohlyjopZ 5.10.0-21-amd64 #1 SMP Debian 5.10.162-1 (2023-01-21) x86_64 GNU/Linux`
- autolab/docker commit hash: a4b85d53ecb60f734b37d922a71a3f32bc3114e0

**Steps To Reproduce**
1. Go to the assessment
2. Click on the score of a problem in "Submission summary"
3. Then 500 error occurs

**Current behavior**
<details><summary>The complete error log is shown below:</summary>
<p>

```bash
--- !ruby/exception:TypeError
message: String can't be coerced into Integer
backtrace:
- "/home/app/webapp/app/controllers/assessments_controller.rb:684:in `+'"
- "/home/app/webapp/app/controllers/assessments_controller.rb:684:in `block in parseScore'"
- "/home/app/webapp/app/controllers/assessments_controller.rb:683:in `each'"
- "/home/app/webapp/app/controllers/assessments_controller.rb:683:in `parseScore'"
- "/home/app/webapp/app/controllers/assessments_controller.rb:632:in `viewFeedback'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_controller/metal/basic_implicit_render.rb:6:in
  `send_action'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/abstract_controller/base.rb:195:in
  `process_action'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_controller/metal/rendering.rb:30:in
  `process_action'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/abstract_controller/callbacks.rb:42:in
  `block in process_action'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/callbacks.rb:112:in
  `block in run_callbacks'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actiontext-6.0.5/lib/action_text/rendering.rb:19:in
  `with_renderer'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actiontext-6.0.5/lib/action_text/engine.rb:55:in
  `block (4 levels) in <class:Engine>'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/callbacks.rb:121:in
  `instance_exec'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/callbacks.rb:121:in
  `block in run_callbacks'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/core_ext/time/zones.rb:66:in
  `use_zone'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/browser-timezone-rails-1.1.0/lib/browser-timezone-rails.rb:22:in
  `set_time_zone'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/callbacks.rb:121:in
  `block in run_callbacks'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/callbacks.rb:139:in
  `run_callbacks'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/abstract_controller/callbacks.rb:41:in
  `process_action'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_controller/metal/rescue.rb:22:in
  `process_action'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_controller/metal/instrumentation.rb:33:in
  `block in process_action'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/notifications.rb:180:in
  `block in instrument'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/notifications/instrumenter.rb:24:in
  `instrument'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/notifications.rb:180:in
  `instrument'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_controller/metal/instrumentation.rb:32:in
  `process_action'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_controller/metal/params_wrapper.rb:245:in
  `process_action'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/activerecord-6.0.5/lib/active_record/railties/controller_runtime.rb:27:in
  `process_action'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/abstract_controller/base.rb:136:in
  `process'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionview-6.0.5/lib/action_view/rendering.rb:39:in
  `process'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_controller/metal.rb:190:in
  `dispatch'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_controller/metal.rb:254:in
  `dispatch'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/routing/route_set.rb:50:in
  `dispatch'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/routing/route_set.rb:33:in
  `serve'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/journey/router.rb:49:in
  `block in serve'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/journey/router.rb:32:in
  `each'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/journey/router.rb:32:in
  `serve'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/routing/route_set.rb:834:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/rack/agent_hooks.rb:30:in
  `traced_call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/rack/browser_monitoring.rb:33:in
  `traced_call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/rack-attack-6.6.1/lib/rack/attack.rb:103:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/exception_notification-4.5.0/lib/exception_notification/rack.rb:49:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/rack-ssl-enforcer-0.2.9/lib/rack/ssl-enforcer.rb:52:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/rack-attack-6.6.1/lib/rack/attack.rb:110:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/warden-1.2.9/lib/warden/manager.rb:36:in `block
  in call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/warden-1.2.9/lib/warden/manager.rb:34:in `catch'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/warden-1.2.9/lib/warden/manager.rb:34:in `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/rack-2.2.6.2/lib/rack/tempfile_reaper.rb:15:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/rack-2.2.6.2/lib/rack/etag.rb:27:in `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/rack-2.2.6.2/lib/rack/conditional_get.rb:27:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/rack-2.2.6.2/lib/rack/head.rb:12:in `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/http/content_security_policy.rb:18:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/rack-2.2.6.2/lib/rack/session/abstract/id.rb:266:in
  `context'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/rack-2.2.6.2/lib/rack/session/abstract/id.rb:260:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/cookies.rb:654:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/callbacks.rb:27:in
  `block in call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/callbacks.rb:101:in
  `run_callbacks'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/callbacks.rb:26:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/actionable_exceptions.rb:18:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/debug_exceptions.rb:32:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/show_exceptions.rb:33:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/railties-6.0.5/lib/rails/rack/logger.rb:37:in
  `call_app'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/railties-6.0.5/lib/rails/rack/logger.rb:26:in
  `block in call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/tagged_logging.rb:80:in
  `block in tagged'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/tagged_logging.rb:28:in
  `tagged'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/tagged_logging.rb:80:in
  `tagged'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/railties-6.0.5/lib/rails/rack/logger.rb:26:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/remote_ip.rb:81:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/request_id.rb:27:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/rack-2.2.6.2/lib/rack/method_override.rb:24:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/rack-2.2.6.2/lib/rack/runtime.rb:22:in `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/cache/strategy/local_cache_middleware.rb:29:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/executor.rb:14:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/static.rb:126:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/rack-2.2.6.2/lib/rack/sendfile.rb:110:in `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/ssl.rb:74:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/host_authorization.rb:97:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/railties-6.0.5/lib/rails/engine.rb:527:in `call'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/railties-6.0.5/lib/rails/railtie.rb:190:in
  `public_send'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/railties-6.0.5/lib/rails/railtie.rb:190:in
  `method_missing'"
- "/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in
  `call'"
- "/usr/lib/ruby/vendor_ruby/phusion_passenger/rack/thread_handler_extension.rb:107:in
  `process_request'"
- "/usr/lib/ruby/vendor_ruby/phusion_passenger/request_handler/thread_handler.rb:149:in
  `accept_and_process_next_request'"
- "/usr/lib/ruby/vendor_ruby/phusion_passenger/request_handler/thread_handler.rb:110:in
  `main_loop'"
- "/usr/lib/ruby/vendor_ruby/phusion_passenger/request_handler.rb:419:in `block (3
  levels) in start_threads'"
- "/usr/lib/ruby/vendor_ruby/phusion_passenger/utils.rb:113:in `block in create_thread_and_abort_on_exception'"
Stack Dump
/home/app/webapp/app/controllers/assessments_controller.rb:684:in `+'
/home/app/webapp/app/controllers/assessments_controller.rb:684:in `block in parseScore'
/home/app/webapp/app/controllers/assessments_controller.rb:683:in `each'
/home/app/webapp/app/controllers/assessments_controller.rb:683:in `parseScore'
/home/app/webapp/app/controllers/assessments_controller.rb:632:in `viewFeedback'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_controller/metal/basic_implicit_render.rb:6:in `send_action'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/abstract_controller/base.rb:195:in `process_action'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_controller/metal/rendering.rb:30:in `process_action'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/abstract_controller/callbacks.rb:42:in `block in process_action'
/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/callbacks.rb:112:in `block in run_callbacks'
/usr/local/rvm/gems/ruby-2.7.7/gems/actiontext-6.0.5/lib/action_text/rendering.rb:19:in `with_renderer'
/usr/local/rvm/gems/ruby-2.7.7/gems/actiontext-6.0.5/lib/action_text/engine.rb:55:in `block (4 levels) in '
/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/callbacks.rb:121:in `instance_exec'
/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/callbacks.rb:121:in `block in run_callbacks'
/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/core_ext/time/zones.rb:66:in `use_zone'
/usr/local/rvm/gems/ruby-2.7.7/gems/browser-timezone-rails-1.1.0/lib/browser-timezone-rails.rb:22:in `set_time_zone'
/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/callbacks.rb:121:in `block in run_callbacks'
/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/callbacks.rb:139:in `run_callbacks'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/abstract_controller/callbacks.rb:41:in `process_action'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_controller/metal/rescue.rb:22:in `process_action'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_controller/metal/instrumentation.rb:33:in `block in process_action'
/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/notifications.rb:180:in `block in instrument'
/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/notifications.rb:180:in `instrument'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_controller/metal/instrumentation.rb:32:in `process_action'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_controller/metal/params_wrapper.rb:245:in `process_action'
/usr/local/rvm/gems/ruby-2.7.7/gems/activerecord-6.0.5/lib/active_record/railties/controller_runtime.rb:27:in `process_action'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/abstract_controller/base.rb:136:in `process'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionview-6.0.5/lib/action_view/rendering.rb:39:in `process'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_controller/metal.rb:190:in `dispatch'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_controller/metal.rb:254:in `dispatch'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/routing/route_set.rb:50:in `dispatch'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/routing/route_set.rb:33:in `serve'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/journey/router.rb:49:in `block in serve'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/journey/router.rb:32:in `each'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/journey/router.rb:32:in `serve'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/routing/route_set.rb:834:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/rack/agent_hooks.rb:30:in `traced_call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/rack/browser_monitoring.rb:33:in `traced_call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/rack-attack-6.6.1/lib/rack/attack.rb:103:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/exception_notification-4.5.0/lib/exception_notification/rack.rb:49:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/rack-ssl-enforcer-0.2.9/lib/rack/ssl-enforcer.rb:52:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/rack-attack-6.6.1/lib/rack/attack.rb:110:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/warden-1.2.9/lib/warden/manager.rb:36:in `block in call'
/usr/local/rvm/gems/ruby-2.7.7/gems/warden-1.2.9/lib/warden/manager.rb:34:in `catch'
/usr/local/rvm/gems/ruby-2.7.7/gems/warden-1.2.9/lib/warden/manager.rb:34:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/rack-2.2.6.2/lib/rack/tempfile_reaper.rb:15:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/rack-2.2.6.2/lib/rack/etag.rb:27:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/rack-2.2.6.2/lib/rack/conditional_get.rb:27:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/rack-2.2.6.2/lib/rack/head.rb:12:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/http/content_security_policy.rb:18:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/rack-2.2.6.2/lib/rack/session/abstract/id.rb:266:in `context'
/usr/local/rvm/gems/ruby-2.7.7/gems/rack-2.2.6.2/lib/rack/session/abstract/id.rb:260:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/cookies.rb:654:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/callbacks.rb:27:in `block in call'
/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/callbacks.rb:101:in `run_callbacks'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/callbacks.rb:26:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/actionable_exceptions.rb:18:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/debug_exceptions.rb:32:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/railties-6.0.5/lib/rails/rack/logger.rb:37:in `call_app'
/usr/local/rvm/gems/ruby-2.7.7/gems/railties-6.0.5/lib/rails/rack/logger.rb:26:in `block in call'
/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/tagged_logging.rb:80:in `block in tagged'
/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/tagged_logging.rb:28:in `tagged'
/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/tagged_logging.rb:80:in `tagged'
/usr/local/rvm/gems/ruby-2.7.7/gems/railties-6.0.5/lib/rails/rack/logger.rb:26:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/request_id.rb:27:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/rack-2.2.6.2/lib/rack/method_override.rb:24:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/rack-2.2.6.2/lib/rack/runtime.rb:22:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/activesupport-6.0.5/lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/executor.rb:14:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/static.rb:126:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/rack-2.2.6.2/lib/rack/sendfile.rb:110:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/ssl.rb:74:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/actionpack-6.0.5/lib/action_dispatch/middleware/host_authorization.rb:97:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/railties-6.0.5/lib/rails/engine.rb:527:in `call'
/usr/local/rvm/gems/ruby-2.7.7/gems/railties-6.0.5/lib/rails/railtie.rb:190:in `public_send'
/usr/local/rvm/gems/ruby-2.7.7/gems/railties-6.0.5/lib/rails/railtie.rb:190:in `method_missing'
/usr/local/rvm/gems/ruby-2.7.7/gems/newrelic_rpm-8.8.0/lib/new_relic/agent/instrumentation/middleware_tracing.rb:100:in `call'
/usr/lib/ruby/vendor_ruby/phusion_passenger/rack/thread_handler_extension.rb:107:in `process_request'
/usr/lib/ruby/vendor_ruby/phusion_passenger/request_handler/thread_handler.rb:149:in `accept_and_process_next_request'
/usr/lib/ruby/vendor_ruby/phusion_passenger/request_handler/thread_handler.rb:110:in `main_loop'
/usr/lib/ruby/vendor_ruby/phusion_passenger/request_handler.rb:419:in `block (3 levels) in start_threads'
/usr/lib/ruby/vendor_ruby/phusion_passenger/utils.rb:113:in `block in create_thread_and_abort_on_exception'
```

</p>
</details>

**Expected behavior**
I actually don't know why the score of a problem is clickable. The autograder outputs the json in the end. This button works fine when the grading is in process, and students can see the autograder outputs. Once grading is complete, 500 error occurs.

In other words, the url `...assessments/lab2/viewFeedback?feedback=23&submission_id=165` shows the error, but `...assessments/lab2/submissions/165/view?header_position=-1` works fine.

**Screenshots**

**Comment by damianhxy** [comment]
Possibly related to #1315, #1718.

@XinyuZeng, I suspect that you have more than one trailing newline after the JSON string -- there should only be one for parsing to work correctly. Could you confirm?

**Comment by XinyuZeng** [comment]
I think there is only one newline in my script, where the last line of python script is `print(json.dumps(result))`.

The last line of Autograder output (viewed on autolab) is like:

![SharedScreenshot](https://user-images.githubusercontent.com/30414224/231335542-00ce7254-c0cb-491b-a65d-ee9ab135b7b9.jpg)

**Comment by damianhxy** [comment]
I think the problem is that your numbers are in quotes. They should be integers, like so:

![Screenshot 2023-04-12 at 00 23 08](https://user-images.githubusercontent.com/9074856/231348943-8c3dcaf4-3233-460c-b8c0-7f5a79f01448.png)

**Comment by XinyuZeng** [comment]
Thanks @damianhxy ! The problem is that the scores in my script is loaded from a rubrics.csv file, so I need to convert the string value into float. Probably add another conversion on Autolab side and provide better error message (e.g., catch exception and print suggestions) can make it more clearer to users. But anyway this issue is fixed :)

**Comment by damianhxy** [comment]
@XinyuZeng Glad to hear that resolved your issue! I've opened a new issue to track progress on your suggestion, and will close this one.

**Comment by damianhxy** [comment]
@XinyuZeng We have added type conversion, so strings will be fine in the future
